.ci/compute_projects.py

```diff
@@ -267,6 +284,12 @@ def get_env_variables(modified_files: list[str], platform: str) -> Set[str]:
     runtimes_check_targets_needs_reconfig = _compute_project_check_targets(
         runtimes_to_test_needs_reconfig
     )
+
+    # Check if both clang and mlir are in projects_to_build to enable CIR
+    enable_cir = (
+        "ON" if "clang" in projects_to_build and "mlir" in projects_to_build else "OFF"
```

Comment:
This is a bit weird. Why aren't we just checking if `CIR` is in modified projects?

Reply:
If no CIR files have been modified, CIR won't be in the modified projects list. This is a bit redundant with the current dependencies, because having mlir in the list causes clang to be added to the list. I think that's an indirect dependency. When you modify mlir, it adds flang to the builds becuase flang is a test dependency of mlir. In turn, flag has a dependency on clang. The check here, however, is more directly checking the conditions under which it is cheap to build CIR.

I'll admit this all feels like a bit of a hack, mostly due to the fact that CIR isn't an independent project that can be part of the `projects_to_build` list.

Reply:
Sure